internal/controllers/topology/cluster/scope/state_test.go

```diff
@@ -155,3 +157,147 @@ func TestUpgrading(t *testing.T) {
 		g.Expect(got).To(BeComparableTo(want))
 	})
 }
+
+func TestMPIsUpgrading(t *testing.T) {
```

Comment:
Sorry didn't get around to review your PRs yet (and I'm soon on PTO).

Just fyi for now. That this will be also added in this PR (and moved around): https://github.com/kubernetes-sigs/cluster-api/pull/10063/files